src/components/CheckIn/PiecesList.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+  Checkbox,
+  MultiColumnList,
+} from '@folio/stripes/components';
+
+const PiecesList = ({ items = [], toggleItem, toggleAll, isAllChecked = false }) => {
+  const resultsFormatter = {
+    'isChecked': piece => (
+      <Checkbox
+        checked={piece.isChecked}
+        type="checkbox"
+        onChange={() => toggleItem(piece)}
+      />
+    ),
+    'title': piece => piece.title,
+    'piece': piece => piece.caption,
+    'supplement': piece => (
+      <Checkbox
+        checked={piece.supplement}
+        disabled
+        type="checkbox"
+      />
+    ),
+    'poLineNumber': piece => piece.poLineNumber,
+    'comment': piece => piece.comment,
+    'pieceStatus': piece => piece.receivingStatus,
+  };
+
+  return (
+    <MultiColumnList
+      contentData={items}
+      formatter={resultsFormatter}
+      visibleColumns={['isChecked', 'title', 'piece', 'supplement', 'poLineNumber', 'comment', 'pieceStatus']}
+      columnMapping={{
+        isChecked: (
+          <Checkbox
+            name="isAllChecked"
+            checked={isAllChecked}
+            type="checkbox"
+            onChange={() => toggleAll()}
+          />
+        ),
+        title: <FormattedMessage id="ui-orders.receiving.title" />,
+        piece: <FormattedMessage id="ui-orders.checkIn.piece" />,
+        supplement: <FormattedMessage id="ui-orders.checkIn.supplement" />,
+        poLineNumber: <FormattedMessage id="ui-orders.receiving.poLine" />,
+        comment: <FormattedMessage id="ui-orders.checkIn.comment" />,
+        pieceStatus: <FormattedMessage id="ui-orders.checkIn.pieceStatus" />,
+      }}
```

Comment:
Is it possible to create something like object instead of separate key/value lines in translation file for preventing redundant calls?

Reply:
I'm not sure that I've got your idea, could you please elaborate more on this? thanks